fetcher-pipeline-api/app/main.py

```diff
@@ -0,0 +1,219 @@
+                    "type": "audio",
+                    "domain": "scripture",
+                    "resourceType": "bible",
+                    "namespace": "audio_biel",
+                    "files": []
+                }
+
+            item = {
+                "size": os.path.getsize(filename),
+                "url": path_to_url(filename),
+                "fileType": Path(filename).suffix[1:],
+                "hash": calc_md5_hash(filename),
+                "isWholeBook": parts.chapter is None,
+                "isWholeProject": False,
+                "bookName": get_book_name(books, parts.book_slug),
+                "bookSlug": parts.book_slug.capitalize(),
+                "chapter": parts.chapter
+            }
+
+            items.append(item)
+
+    messages = []
+
+    if message is not None:
+        # chunks are done in about this size because azure service bus has a 256kb  size limit, and unchunked a nt in all file types and qualities is 3000+ files, would be over limit. Some prelim testing saw this number of urls consistently come in around 225 kb give or take a little. 
+        chunks = split_array(items, 800)
+        print(len(chunks))
+        for chunk in chunks:
+            chunk_message = message.copy()
+            chunk_message["files"] = chunk
+            messages.append(chunk_message)
+
+        await send_messages(messages)
+
+    return messages
+
+
+def extract_parts(path_str: str) -> Parts:
+    # dependent on cdn file structure
+    parts = re.search(re.escape(CONTENT_DIR) + r"/(.+?)/(.+?)/(.+?)(?:/(\d+))?/.*?", path_str)
+    return Parts(
+        language_id=parts.group(1),
+        resource_id=parts.group(2),
+        book_slug=parts.group(3),
+        chapter=None if parts.group(4) is None else parts.group(4)
+    )
+
+
+def calc_md5_hash(file_path) -> str:
+    with open(file_path) as file, mmap(file.fileno(), 0, access=ACCESS_READ) as file:
+        return md5(file).hexdigest()
+
+
+def get_book_name(books, slug) -> str:
+    book = next((sub for sub in books if sub["slug"] == slug), None)
+    return book["name"] if book is not None else slug.capitalize()
+
+
+def path_to_url(file_path) -> str:
+    return urljoin(CONTENT_URL, file_path)
+
+
+def split_array(in_array, size):
+    return [in_array[i:i + size] for i in range(0, len(in_array), size)]
+
+
+async def send_messages(messages):
+    async with ServiceBusClient.from_connection_string(
+        conn_str=NAMESPACE_CONNECTION_STR,
+        logging_enable=True
+    ) as service_bus_client:
+        sender = service_bus_client.get_queue_sender(queue_name=QUEUE_NAME)
+        async with sender:
+            batch_message = await sender.create_message_batch()
+            for message in messages:
+                try:
+                    batch_message.add_message(ServiceBusMessage(json.dumps(message)))
+                except ValueError:
+                    break
+
+                await sender.send_messages(batch_message)
```

Comment:
This should be indented one level up to run after `for`loop, not in the loop. My bad.